code/core/system/proxy-velodyne64/src/ProxyVelodyne64.cpp

```diff
@@ -31,35 +32,48 @@ namespace opendlv {
 namespace core {
 namespace system {
 namespace proxy {
+using namespace odcore::wrapper;
+using namespace odcore::io::udp;
 
-    using namespace odcore::wrapper;
-    using namespace odcore::io::udp;
+ProxyVelodyne64::ProxyVelodyne64(const int &argc, char **argv)
+    : TimeTriggeredConferenceClientModule(argc, argv, "ProxyVelodyne64")
+    , memoryName()
+    , memorySize(0)
+    , udpReceiverIP()
+    , udpPort(0)
+    , VelodyneSharedMemory(NULL)
+    , udpreceiver(NULL)
+    , v64dSp(NULL) {}
 
-    ProxyVelodyne64::ProxyVelodyne64(const int &argc, char **argv)
-        : TimeTriggeredConferenceClientModule(argc, argv, "ProxyVelodyne64"),
-            VelodyneSharedMemory(SharedMemoryFactory::createSharedMemory(NAME, SIZE)),
-            udpreceiver(UDPFactory::createUDPReceiver(RECEIVER, PORT)),
-            v64d(VelodyneSharedMemory,getConference()){}
+ProxyVelodyne64::~ProxyVelodyne64() {}
 
-    ProxyVelodyne64::~ProxyVelodyne64() {}
+void ProxyVelodyne64::setUp() {
+    memoryName = getKeyValueConfiguration().getValue< string >("ProxyVelodyne64.sharedMemory.name");
+    memorySize = getKeyValueConfiguration().getValue< uint32_t >("ProxyVelodyne64.sharedMemory.size");
+    VelodyneSharedMemory = SharedMemoryFactory::createSharedMemory(memoryName, memorySize);
 
-    void ProxyVelodyne64::setUp() {
-        udpreceiver->setStringListener(&v64d);
-        // Start receiving bytes.
-        udpreceiver->start();
-    }
+    udpReceiverIP = getKeyValueConfiguration().getValue< string >("ProxyVelodyne64.udpReceiverIP");
+    udpPort = getKeyValueConfiguration().getValue< uint32_t >("ProxyVelodyne64.udpPort");
+    udpreceiver = UDPFactory::createUDPReceiver(udpReceiverIP, udpPort);
 
-    void ProxyVelodyne64::tearDown() {
-        udpreceiver->stop();
-        udpreceiver->setStringListener(NULL);
-    }
+    v64dSp = shared_ptr< velodyne64Decoder >(new velodyne64Decoder(VelodyneSharedMemory, getConference(), getKeyValueConfiguration().getValue< string >("ProxyVelodyne64.calibration")));
 
-    // This method will do the main data processing job.
-    odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode ProxyVelodyne64::body() {
-        while (getModuleStateAndWaitForRemainingTimeInTimeslice() == odcore::data::dmcp::ModuleStateMessage::RUNNING) {
-        }
-        return odcore::data::dmcp::ModuleExitCodeMessage::OKAY;
+    udpreceiver->setStringListener(v64dSp.get());
+    // Start receiving bytes.
+    udpreceiver->start();
+}
+
+void ProxyVelodyne64::tearDown() {
+    udpreceiver->stop();
+    udpreceiver->setStringListener(NULL);
+}
+
+// This method will do the main data processing job.
+odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode ProxyVelodyne64::body() {
```

Comment:
I have changed both proxy-velodyne16 and 64 to data-triggered. However, I am forced to implement the method "void ProxyVelodyne16::nextContainer(odcore::data::Container &c)", and I am forced to do something with "c", though we do not need this method. Currently I print out the container type. This may not be a good idea because it keeps printing container types at high speed. If it is time-triggered with empty loop, by default we have frequency 1, which is much lower frequency. Is data-triggered still better in our case?